tensorly/decomposition/tests/test_parafac2.py

```diff
@@ -0,0 +1,172 @@
+import itertools
+
+import numpy as np
+import pytest
+
+import tensorly as tl
+from ...random import check_random_state, random_parafac2
+from ... import backend as T
+from ...testing import assert_array_equal, assert_
+from ..parafac2 import parafac2, initialize_decomposition
+from ...parafac2_tensor import parafac2_to_tensor, parafac2_to_slices
+
+
+def corrcoef(A, B):
+    Ac = A - T.mean(A, axis=0)
+    Bc = B - T.mean(B, axis=0)
+
+    As = Ac/T.norm(Ac, axis=0)
+    Bs = Bc/T.norm(Bc, axis=0)
+
+    return T.dot(T.transpose(As), Bs)
+
+
+def best_correlation(A, B):
+    _, r = T.shape(A)
+    corr_matrix = T.abs(corrcoef(A, B))
+
+    best_corr = 0
+    for permutation in itertools.permutations(range(r)):
+        corr = 1
+        for i, j in zip(range(r), permutation):
+            corr *= corr_matrix[i, j]
+        
+        if corr > best_corr:
+            best_corr = corr
+    
+    return best_corr
+
+
+@pytest.mark.parametrize("normalize_factors", [True, False])
+def test_parafac2(normalize_factors):
+    rng = check_random_state(1234)
+    tol_norm_2 = 10e-2
+    rank = 6
+
+    random_parafac2_tensor = random_parafac2(
+        shapes=[(50 + rng.randint(10),60) for _ in range(40)],
+        rank=rank,
+        random_state=rng,
+    )
+    # It is difficult to correctly identify B[i, :, r] if A[i, r] is small.
+    # This is sensible, since then B[i, :, r] contributes little to the total value of X.
+    # To test the PARAFAC2 decomposition in the precence of roundoff errors, we therefore add
+    # 0.01 to the A factor matrix.
+    random_parafac2_tensor.factors[0] += 0.01
```

Comment:
Since it is just a test we don't really care about the memory (the factor is small anyway)..
```suggestion
    random_parafac2_tensor.factors[0] = random_parafac2_tensor.factors[0] + 0.01
```